eng/pipelines/variables/common.yml

```diff
@@ -10,14 +10,18 @@ variables:
   value: false
 - name: publicGitRepoUri
   value: https://github.com/dotnet/dotnet-buildtools-prereqs-docker
-- name: manifestVariables
-  value: --var UniqueId=$(sourceBuildId)
+- ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/production') }}:
+  - name: manifestVariables
+    value: --var UniqueId=$(sourceBuildId) --var FloatingTagSuffix=latest
+- ${{ else }}:
+  - name: manifestVariables
+    value: --var UniqueId=$(sourceBuildId) --var FloatingTagSuffix=staging
```

Comment:
Will this mean that any branch other than production built officially will get the staging tag? Should this be under `- ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/main') }}:` instead?

Reply:
My understanding is that the pipelines that publish only trigger on main, and not anything else. I guess it's possible someone could manually run the pipeline and that could screw this logic up some.

This brings up a different question for me though: I will need to update the triggers to include the production branch.

Reply:
Yep. I would make else this for main. People do ocassionally run custom official builds. We had this situation with dotnet/installer in the 3.1 timeframe. Custom builds would overwrite the latest tags for the SDK.

Reply:
We will need to have this be set as something, though, because otherwise we'd end up with some broken code, I think. Maybe Build.SourceBranchName?

Reply:
Should be $(Build.SourceBranch)? Somewhat annoyingly, `Build.SourceBranchName` only takes the last element of the path. For instance, if the branch name were `release/6.0`, `SourceBranchName` is `6.0`:

https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables-devops-services

Reply:
SourceBranch also adds refs.heads too.

Reply:
We can't have the slashes in it, or I *think* docker will get confused.